apps/src/StudioApp.js

```diff
@@ -1810,6 +1810,19 @@ StudioApp.prototype.getUnfilledFunctionalBlockError = function (topLevelType) {
   }
 };
 
+/**
+ * @returns {boolean} True if we have a function or variable named "" (empty string)
+ */
+StudioApp.prototype.hasEmptyFunctionOrVariableName = function () {
+  return Blockly.mainBlockSpace.getTopBlocks().some(function (block) {
+    if (block.type !== 'functional_definition') {
+      return false;
+    }
+
+    return block.getProcedureInfo().name === '';
```

Comment:
Second, it looks like `block.getProcedureInfo().name` will sometimes be `null` if the title does not exist:

```javascript
// from getProcedureInfo() in procedures.js and functional_procedures.js
name: this.getTitleValue('NAME')

// from getTitleValue() in block.js
Blockly.Block.prototype.getTitleValue = function(name) {
  var title = this.getTitle_(name);
  if (title) {
    return title.getValue();
  }
  return null;
};
```

So this check probably needs to return true for `null` in addition to empty string.  I suggest just doing a boolean coercion `return !!(block.getProcedureInfo().name)`, which also seems to be faster than the string comparison, possibly by 25% for many iterations :grin:.